### PR TITLE
Update First() to manage QueryableTableValue

### PR DESCRIPTION
**Pull request by LucGenetier**
Update First() to manage QueryableTableValue

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj [hunk 1]
```diff
@@ -7,6 +7,11 @@
     <Configurations>Debug;Release</Configurations>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj:10]: Why this change? Seems unrelated.

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj:10]: Right. Was missing symbols, so I fixed it at the same time

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@
 using Microsoft.PowerFx.Core.IR;
 using Microsoft.PowerFx.Interpreter;
 using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Syntax.PrettyPrintVisitor;
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs:11]: unused using?

File: src/libraries/Microsoft.PowerFx.Interpreter/PowerFxInterpreterException.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    [Serializable]
+    public class PowerFxInterpreterException : Exception
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/PowerFxInterpreterException.cs:7]: Is the expectation that this be thrown from a derived table value?

File: src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj (hunk 1 above)
Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj:10]: shouldn't we always do this, even for retail?

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj:10]: Core, yes

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj:10]: Let me remove that from this PR and create a new one with a change in Directory.Build.Props

Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/Microsoft.PowerFx.Interpreter.csproj]: [https://github.com/microsoft/Power-Fx/pull/606](https://github.com/microsoft/Power-Fx/pull/606)

File: src/libraries/Microsoft.PowerFx.Interpreter/PowerFxInterpreterException.cs (hunk 3 above)
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/PowerFxInterpreterException.cs:7]: Absolutely

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — CHANGES_REQUESTED** [review]
:clock1:

File: src/libraries/Microsoft.PowerFx.Interpreter/PowerFxInterpreterException.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.Serialization;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    [Serializable]
+    public class PowerFxInterpreterException : Exception
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/PowerFxInterpreterException.cs:10]: >PowerFxInterpreterException [](http://example.com/codeflow?start=17&length=27)

use DValue<T> instead.

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**
